src/CTA.WebForms2Blazor/ControlConverters/ControlConverter.cs

```diff
@@ -64,10 +113,48 @@ public static HtmlDocument ConvertEmbeddedCode(HtmlDocument document)
 
             //Not implemented/used yet
             //htmlString = EmbeddedCodeReplacers.AspExpRegex.Replace(htmlString, aspExprEval);
+            
+            return htmlString;
+        }
+        
+        //This function only updates the first child node that matches the name,
+        //but for current purposes there should only be one matching node
+        protected bool UpdateInnerHtmlNode(HtmlNode outerNode, string targetName, 
+            string template = null, 
+            string newName = null, 
+            IEnumerable<Attribute> newAttributes = null, 
+            string newBody = null)
+        {
+            var lowerName = targetName.ToLower();
+
+            //Ideally use .SelectSingleNode, but doesnt work with names with ':' character and is less flexible
+            //var selectedNode = node.SelectSingleNode(lowerName);
+            
+            var selectedNodes = outerNode.Descendants().Where(child =>
+            {
+                return child.Name.ToLower() == lowerName;
+            }).ToList();
+
+            if (selectedNodes.Count > 0)
+            {
+                template??= NodeTemplate;
+                newName??= targetName;
+                newAttributes??= new List<Attribute>();
```

Comment:
Seems like `newAttributes` is only used for the `GetNewAttributes()` call below, that function sets its corresponding `additionalAttributes` parameter to a new list if the parameter value is none so this line is likely not needed, especially if you decide to make the modification I suggested above in an effort to prevent creation of collections we won't ever use